pkg/etcd-backup-operator/controller/s3_backup.go

```diff
@@ -0,0 +1,77 @@
+package controller
+
+import (
+	"context"
+	"fmt"
+	"io/ioutil"
+	"os"
+
+	"github.com/aws/aws-sdk-go/aws/session"
+	api "github.com/coreos/etcd-operator/pkg/apis/etcd/v1beta2"
+	"github.com/coreos/etcd-operator/pkg/backup"
+	"github.com/coreos/etcd-operator/pkg/backup/backupapi"
+	backupS3 "github.com/coreos/etcd-operator/pkg/backup/s3"
+	"github.com/coreos/etcd-operator/pkg/cluster/backupstorage"
+	"github.com/coreos/etcd-operator/pkg/util/constants"
+)
+
+const (
+	tmp = "/tmp"
+)
+
+// TODO: remove this and use backend interface for other options (PV, Azure)
+func (b *Backup) handleS3(clusterName string, s3 *api.S3Source) error {
+	ctx, cancel := context.WithTimeout(context.Background(), constants.DefaultRequestTimeout)
+	rc, version, rev, err := backup.GetSnap(ctx, b.kubecli, nil, b.namespace, clusterName)
+	cancel()
+	if err != nil {
+		return fmt.Errorf("failed to retrieve snapshot: (%v)", err)
+	}
+
+	awsDir, so, err := b.setupAWSConfig(s3.AWSSecret)
+	if err != nil {
+		return fmt.Errorf("failed to set up aws config: (%v)", err)
+	}
+	defer os.RemoveAll(awsDir)
+
+	prefix := backupapi.ToS3Prefix(s3.Prefix, b.namespace, clusterName)
+	s3Dir, be, err := b.makeS3Backend(so, prefix, s3.S3Bucket)
+	if err != nil {
+		return fmt.Errorf("failed to create s3 backend: (%v)", err)
+	}
+	defer os.RemoveAll(s3Dir)
+
+	_, err = backup.WriteSnap(version, rev, be, rc)
+	if err != nil {
+		return fmt.Errorf("failed to write snap to s3 backend: (%v)", err)
+	}
+	return err
+}
+
+func (b *Backup) setupAWSConfig(secret string) (awsDir string, so *session.Options, err error) {
```

Comment:
Any reason to use named return variables? We're not doing a naked return here and since we're not using named return variables anywhere else it might be best to not name them to stay consistent.

Reply:
@hasbro17 I hope that the named variables give some context on what to return. but I can remove them if needed.

Reply:
Sure I get it. But it's probably best to not name them since that's the convention we're using through the rest of the codebase.
For most functions, it's already clear to understand from the parameter names of the return statement.
Like `awsDir`. And generally the type tells us more than the abbreviated name anyway, like `so` vs `*session.Options`.

Reply:
sounds good.